Add mcuxpresso exporter

JojoS62 opened the pull request:
## Description
Add an exporter for MCUXpresso (NXP), successor of LPCXpresso

## Status
**IN DEVELOPMENT**
needs work
- base is gnuarmecplipse exporter
- Works for the targets that LPCXpresso supported + LPC54114
- LPC54608 compiles as well, but the MCU settings for debugging are not yet available.
- Supports debug and release builds, develop not yet working
- flags are read from mbed build system and converted to eclipse tools settings


## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

NO APIs changed


## Related PRs
#4819


## Todos
- [x] Tests
- [x] Documentation


## Deploy notes
needs installation of MCUXpresso, tested with version 10.0.2

## Steps to test or reproduce
create a project file with 'mbed export -i mcuxpresso'
import the created project.zip 
** Note: this version always creates a zip file for faster testing. Needs a clean solution for creating a zip file as an option.

mmahadevan108 committed:
Add support for MCUXpresso inside mbed exporter

Signed-off-by: Mahadevan Mahesh <Mahesh.Mahadevan@nxp.com>

JojoS62 committed:
exporter based on lpc54114 template from nxpmicro

- used LPC4088 for testing
- split template for LPC54114 into common and Cortex-Mx dependet parts like in old LPCXpresso exporter
- changed fixed linker script name to name from options
Todo: use flags and configurations from options (like in gae exporter)

JojoS62 committed:
changed template to gae exporter style

creates output for debug, develop and release builds
TODO. replaces some fixed flags, develop build not working , linker has no input files

JojoS62 committed:
make excluded_folders POSIX conform

change backslash to forward slash in excluded_folders set

JojoS62 committed:
created project compiles in MCUX

project is read, but still fixes necessary:
- archticture is fixed entry, read from config
- linker options
- linker script file ?
- linker settings 'managed build' page causes error
- develop still not working

JojoS62 committed:
added generated flags

target family, fpu
removed 'develop' proffile, it causes errors in import

JojoS62 committed:
added templates for LPC4088 and LPC11H37

target specific templates use a  common template .cproject.tmpl and extend it by adding target information. This is necessary for the flash and debugging tools

JojoS62 committed:
added targets from LPCXpresso exporter

added all targets from older LPCXpresso exporter

JojoS62 committed:
use optimization setting from flags

optimization is set for debug and release builds

JojoS62 committed:
fixed linker errors

for mbed 2: user lib (e.g. mbed lib) needs to be added to library group
linker command changed: add flags after object files

JojoS62 committed:
added support for LPC54114 and LPC54608

project template for LPC54114 is from preinstalled MCU, template for LPC54608 is empty because SDK is not yet available

JojoS62 committed:
added missing flags

added:
- compiler standard
- optimization flags
- warning flags
- different other flags

JojoS62 committed:
removed unnecessary entrys

theotherjimmy reviewed (changes requested):
It looks like you copied the contents of the gnuarmeclipse exporter. To prevent duplication of fixes, could you import it and override things instead?

JojoS62 committed:
derive from gnuarmeclipse class

derive from gnuarmeclipse class instead of copy
is_target_supported is true when .cproject template for targetname exists

JojoS62 committed:
revert hack to force zipfile creation

remove forcing to create zipfiles

JojoS62 committed:
fixed is_target_supported()

path for checking the templates was not independent

theotherjimmy reviewed (changes requested):
This is looking really good. I have left a few requests and questions below. I would appreciate it if you could answer or address them as appropriate.

JojoS62 committed:
remove test modification

changed LPC11U37H_401 back to original release_versions [2] only

JojoS62 committed:
add template for LPC1549

JojoS62 committed:
use UID class from gnuarmeclipse exporter

use global UID class
removed  outcommented code

JojoS62 committed:
removed empty project

was just an copy template for new targets

JojoS62 committed:
added target LPCPC54608

from patch by mmahadevan108

theotherjimmy reviewed (approved):
Looks great. Thanks for addressing my comments.

JojoS62 committed:
changed execuatable name for tests

theotherjimmy reviewed (approved):
Looks great! Let's get this tested and merged!

theotherjimmy reviewed (changes requested):
I'm going to put a "red x" on this PR so that we know not to merge it yet. I'll look into what's causing the lack of testing and push a commit that corrects it, if I get the chance.

JojoS62 committed:
applied patch from mmahadevan108

- fixes invalid char '#' in linker script
- added 'mbedclean'

JojoS62 committed:
add support for K64F target

added template for K64F

JojoS62 committed:
added support for targets with extra SDK

some targets need an additional SDK.

theotherjimmy committed:
Add imports for build

theotherjimmy committed:
Allow eclipse to fail for mysteriously in build
